components/humantask/org.wso2.carbon.humantask/src/main/java/org/wso2/carbon/humantask/core/scheduler/NotificationScheduler.java

```diff
@@ -0,0 +1,272 @@
+                        .item(0).getTextContent();
+            } catch (Exception e) {
+                log.error("Error while evaluating rendering xpath. Please review xpath and deploy " +
+                        "task again.", e);
+            }
+        } else {
+            log.warn("Rendering type " + rendering + "Not found for task definition.");
+        }
+
+        dynamicPropertiesForEmail.put("email.address", mail_to);
+        dynamicPropertiesForEmail.put("email.subject", mail_subject);
+
+
+        return dynamicPropertiesForEmail;
+    }
+
+    public String getMessageOfSmsNotification(HumanTaskBaseConfiguration taskConfiguration, TaskCreationContext context,
+                                              TaskDAO task) {
+
+        smsBody = evaluateTextForXPath(smsBodyTagVal, taskConfiguration, context, task);
+        return smsBody;
+    }
+
+    public String getMessageOfEmailNotification(HumanTaskBaseConfiguration taskConfiguration, TaskCreationContext context,
+                                                TaskDAO task) {
+        emailBody = evaluateTextForXPath(emailBodyTagVal, taskConfiguration, context,
+                task);
+        return emailBody;
+    }
+
+    public String evaluateTextForXPath(String xmlString,
+                                       HumanTaskBaseConfiguration taskConfiguration, TaskCreationContext creationContext,
+                                       TaskDAO task) {
+        String expressionLanguage = taskConfiguration.getExpressionLanguage();
+        ExpressionLanguageRuntime expLangRuntime = HumanTaskServiceComponent.getHumanTaskServer()
+                .getTaskEngine().getExpressionLanguageRuntime(expressionLanguage);
+        EvaluationContext evalContext = creationContext.getEvalContext();
+        String processedString = resolveJAXP(xmlString, task);
+        String[] split = processedString.split("\\s+");
+        StringBuilder builder = new StringBuilder();
+        for (String element : split) {
+            if (element.startsWith(prefix + ":")) {
+                builder.append(" " + expLangRuntime.evaluateAsString(element, evalContext));
+            } else {
+                builder.append(" " + element);
+            }
+        }
+
+        return builder.toString();
+    }
+
+    public String resolveJAXP(String xmlString, TaskDAO task) {
+        String replacedXmlString = CommonTaskUtil.replaceUsingPresentationParams(
+                task.getPresentationParameters(), xmlString);
+        return replacedXmlString;
+    }
+
+    public Document xmlRead(String xmlString) {
+        Document doc = null;
+        try {
+            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder builder = factory.newDocumentBuilder();
+            doc = builder.parse(new InputSource(new StringReader(xmlString)));
+            doc.getDocumentElement().normalize();
+
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Logged the exception in commit https://github.com/wso2/carbon-business-process/commit/c037c2607734679028dc96c59761857176b6dfdb